From 9691188d6bd16d23db049fd769c9371365541d6e Mon Sep 17 00:00:00 2001
From: James Slagle <jslagle@redhat.com>
Date: Thu, 6 Feb 2014 12:30:45 -0500
Subject: [PATCH 05/12] Fix glance package install config

The package install for glance drops the needed paste ini config under
/usr/share/glance, so we need to symlink these files under /etc/glance.

Change-Id: I117c6a1440c3adaf04cd191390a921819a6fcbd7
---
 elements/glance/install.d/glance-package-install/75-glance | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/elements/glance/install.d/glance-package-install/75-glance b/elements/glance/install.d/glance-package-install/75-glance
index 641cc16..2690a4d 100755
--- a/elements/glance/install.d/glance-package-install/75-glance
+++ b/elements/glance/install.d/glance-package-install/75-glance
@@ -10,3 +10,16 @@ install-packages openstack-glance
 touch /var/log/glance/registry.log
 touch /var/log/glance/api.log
 chown -R glance.glance /var/log/glance
+
+DISTRO=`lsb_release -si` || true
+if [[ "RedHatEnterpriseServer CentOS Fedora" =~ "$DISTRO" ]]; then
+    # The package installs for these variants drop their config under
+    # /usr/share/glance.  We need the paste ini files symlinked under
+    # /etc/glance, and the conf files symlinked under /usr/share/glance.
+    ln -s /usr/share/glance/glance-api-dist-paste.ini /etc/glance/glance-api-paste.ini
+    ln -s /usr/share/glance/glance-registry-dist-paste.ini /etc/glance/glance-registry-paste.ini
+    ln -f -s /etc/glance/glance-api.conf /usr/share/glance/glance-api-dist.conf
+    ln -f -s /etc/glance/glance-cache.conf /usr/share/glance/glance-cache-dist.conf
+    ln -f -s /etc/glance/glance-registry.conf /usr/share/glance/glance-registry-dist.conf
+    ln -f -s /etc/glance/glance-scrubber.conf /usr/share/glance/glance-scrubber-dist.conf
+fi
-- 
1.8.5.3

